zig-dxread/src/main.zig

```diff
@@ -0,0 +1,72 @@
+const std = @import("std");
+const fs = std.fs;
+const gpa = std.heap.GeneralPurposeAllocator(.{}){};
```

Comment:
I'd recommend keeping allocators out of global scope, and moving the GPA down to wherever you use it

Reply:
interesting. i was tempted to keep them in global scope as i see it more as a config for now (I dont see myself changing allocators in a program yet and if I do, I would gradually add their choices in).
I'll pass them around, and if i need to pass larger information, ill pass a struct with this information